### PR TITLE
pkg/ipmi: Interpret IPMI command error codes

### PR DESCRIPTION
**Pull request by newgrp**
Currently IPMI command errors manifest as `"invalid response, expected first byte of response to be 0, got: %v"`. This commit changes the error message to depend on the value of the completion code, clarifying the nature of the error to a casual user.

### CODE REVIEW

File: pkg/ipmi/dev.go [hunk 1]
```diff
@@ -13,6 +13,69 @@ import (
 	"golang.org/x/sys/unix"
 )
 
+func (cc CompletionCode) Error() string {
+	switch cc {
+	case IPMI_CC_OK:
```
Comment (hugelgupf) [inline pkg/ipmi/dev.go:18]: this would be good as a map

map[CompletionCode]string

File: pkg/ipmi/dev.go [hunk 2]
```diff
@@ -13,6 +13,69 @@ import (
 	"golang.org/x/sys/unix"
 )
 
+func (cc CompletionCode) Error() string {
```
Comment (hugelgupf) [inline pkg/ipmi/dev.go:16]: It's a little non-idiomatic to make a CompletionCode an error. For one, an error should always be declared as `...Error`.

Here's what I suggest:

```go
func (cc CompletionCode) String() string

type CompletionError CompletionCode

func (ce CompletionError) Error() string {
  return fmt.Sprintf("command completed with non-OK code: %s", CompletionCode(ce))
}
```

and then below, you can

```
rerr = fmt.Errorf("invalid response: %w", CompletionError(cc))
```

when you get the response, for testing you can do e.g. `errors.Is(err, ipmi.CompletionError(ipmi.IPMI_CC_TIMEOUT))`

File: pkg/ipmi/dev.go (hunk 1 above)
Comment (newgrp) [reply pkg/ipmi/dev.go]: Done.

File: pkg/ipmi/dev.go (hunk 2 above)
Comment (newgrp) [reply pkg/ipmi/dev.go]: Done.

Comment (hugelgupf) [reply pkg/ipmi/dev.go]: Missing the rerr = fmt.Errorf part

**Comment by newgrp** [comment]
FYI, I got an email with another comment, but I can't find it in the UI. The UI tells me that "All conversations are resolved" right now.

Also, is the codecov/patch failure a blocking issue? I can add tests for the error strings if it's important, but I'm not sure it adds much real value.

**Comment by hugelgupf** [comment]
> FYI, I got an email with another comment, but I can't find it in the UI. The UI tells me that "All conversations are resolved" right now.
> 
> Also, is the codecov/patch failure a blocking issue? I can add tests for the error strings if it's important, but I'm not sure it adds much real value.

yeah looks like I replied to a resolved conversation. you can open those in the UI. I unresolved it tho so it should show now.

File: pkg/ipmi/dev.go (hunk 2 above)
Comment (newgrp) [reply pkg/ipmi/dev.go]: Ah, I deliberately left that out. We already add context to the error in `CompletionError`'s `Error()` method. I don't think we need to wrap it again when returning it.

**Pull request merged**